mscore/scoreview.cpp

```diff
@@ -1779,81 +1779,88 @@ void ScoreView::cmd(const QAction* a)
 
 void ScoreView::cmd(const char* s)
       {
+      struct ScoreViewCmd {
+            std::vector<const char*> commands;
+            std::function<void()> exec;
+            };
+
       const QByteArray cmd(s);
 
       shadowNote->setVisible(false);
       if (MScore::debugMode)
             qDebug("ScoreView::cmd <%s>", s);
 
-      if (cmd == "escape")
+      const std::vector<ScoreViewCmd> cmdList {
```

Comment:
I suggest that this be declared `static`.

Reply:
Good call.

Reply:
Actually, no, this cannot be static, because the lambda functions need to capture the value of `this` on every execution.

Reply:
We can make these functions accept `ScoreView*` as its parameter, not as captured `this` variable. This would allow not to create the entire `vector` with functions on each `ScoreView::cmd()` invocation.

Reply:
Also maybe this could be made a `map` for faster lookup, not sure how significant would be the effect though.

Reply:
The vector in `Score::cmd()` is not static. Why should this one be any different?

Reply:
Then it would probably be better to restructure `Score::cmd()` as well. Creating large vectors with absolutely equal data each time user invokes a shortcut doesn't seem to be a good idea.

Reply:
> Also maybe this could be made a `map` for faster lookup, not sure how significant would be the effect though.

I had actually considered that as well, but it's can't be made into a simple map due to the nested nature of it (due to the presence of multiple commands mapping to the same lambda).

It can be converted into a two-phase lookup by separating out the cases that have multiple commands, but just as you said, I'm not sure it would be worth the effort.

Reply:
The `cmdList` vector can be changed like this: https://github.com/musescore/MuseScore/commit/3ac32e57c1cc02cde594ded7b2b9df5e62c02980.

Then, since I will have edited almost every line of code anyway, I might as well go ahead and use the proper indentation, which I did not do originally for 2 reasons:
1. I wanted to preserve git blame, which is no longer possible.
2. The resulting diff looks like this: https://github.com/musescore/MuseScore/compare/master...mattmcclinch:restructure-scoreview-cmd-2

Finally, I can do the same for the `cmdList` vector in `Score::cmd()`, as part of this PR or a separate one.

I can push those changes to this branch, either squashing the commits or leaving them unsquashed, whatever you think is best.

Reply:
Much better. I understand the motivation behind not wanting to change the blame attribution, but sometimes you gotta do what you gotta do.

I haven't investigated this enough to know for sure, but I feel like a real fix for this would be to have an `enum` of commands and a globally accessible map of literal strings to `enum` values. Then we could use the `enum` values everywhere internally and just use simple `switch` statements for functions like this.

But this will definitely do for now.

Reply:
Looks good to me either. @mattmcclinch, could you please push the changes to this PR? Separate commits like in `restructure-scoreview-cmd-2` branch seem to be a better option here as they make more clear which exactly changes were going on.